resources/src/operation/mod.rs

```diff
@@ -398,132 +429,140 @@ impl<'de> Deserialize<'de> for Operation {
                 trustee: Some(trustee),
                 ..
             } => {
-                if operation_type == "allow_trust" {
-                    let asset = AssetIdentifier::new(&asset_type, rep.asset_code, rep.asset_issuer)
-                        .map_err(|_| {
-                            de::Error::custom("Code and issuer are required for non-native assets")
-                        })?;
-                    Some(OperationDetail::AllowTrust(
-                        self::allow_trust::AllowTrust::new(trustee, trustor, asset, authorize),
-                    ))
-                } else {
-                    None
-                }
+                let asset = AssetIdentifier::new(&asset_type, asset_code, asset_issuer)
+                    .map_err(|err| de::Error::custom(err))?;
+                Kind::AllowTrust(AllowTrust::new(trustee, trustor, asset, authorize))
             }
-            IntermediateOperation {
-                operation_type,
+            Intermediate {
+                kind: "allow_trust",
+                ..
+            } => {
+                return Err(de::Error::custom(
+                    "Missing fields for allow trust operation.",
+                ))
+            }
+
+            Intermediate {
+                kind: "account_merge",
                 account: Some(account),
                 into: Some(into),
                 ..
+            } => Kind::AccountMerge(AccountMerge::new(account, into)),
+
+            Intermediate {
+                kind: "account_merge",
+                ..
             } => {
-                if operation_type == "account_merge" {
-                    Some(OperationDetail::AccountMerge(
-                        self::account_merge::AccountMerge::new(account, into),
-                    ))
-                } else {
-                    None
-                }
+                return Err(de::Error::custom(
+                    "Missing fields for account merge operation.",
+                ))
             }
-            IntermediateOperation {
-                operation_type,
+
+            Intermediate {
+                kind: "manage_data",
                 name: Some(name),
                 value: Some(value),
                 ..
+            } => Kind::ManageData(ManageData::new(name, value)),
+
+            Intermediate {
+                kind: "manage_data",
+                ..
             } => {
-                if operation_type == "manage_data" {
-                    Some(OperationDetail::ManageData(
-                        self::manage_data::ManageData::new(name, value),
-                    ))
-                } else {
-                    None
-                }
-            }
-            _ => {
-                if &rep.operation_type[..] == "inflation" {
-                    Some(OperationDetail::Inflation)
-                } else {
-                    None
-                }
+                return Err(de::Error::custom(
```

Comment:
I'm debating moving all the errors to the bottom of the match. Thoughts?

Reply:
I like how it is now because it offers additional information about the kind it was expecting.  If we move it down here, maybe we should include the data from Intermediate in the message.

Reply:
Sorry, I meant moving the clauses down, not removing the extra error messaging.